common/src/main/java/org/apache/rocketmq/common/message/MessageClientIDSetter.java

```diff
@@ -37,13 +37,10 @@
         } catch (Exception e) {
             ip = createFakeIP();
         }
-        LEN = ip.length + 2 + 4 + 4 + 2;
-        ByteBuffer tempBuffer = ByteBuffer.allocate(ip.length + 2 + 4);
-        tempBuffer.position(0);
+        LEN = ip.length + 4 + 4 + 4 + 2;
+        ByteBuffer tempBuffer = ByteBuffer.allocate(ip.length + 4 + 4);
         tempBuffer.put(ip);
-        tempBuffer.position(ip.length);
         tempBuffer.putInt(UtilAll.getPid());
-        tempBuffer.position(ip.length + 2);
```

Comment:
pid is short，so maybe putShort can resolve this issue，and this indeed a bug, ip.length + 2 will keep the lower 2 bytes.

Reply:
"The pid_t data type is a signed integer"
https://www.gnu.org/software/libc/manual/html_node/Process-Identification.html

Reply:
https://unix.stackexchange.com/questions/16883/what-is-the-maximum-value-of-the-process-id

Reply:
issue is here：https://github.com/apache/rocketmq/issues/1751

Reply:
and there is a redundant "org.apache.rocketmq.common.MixAll#getPID"